src/main/java/com/exadel/discount/service/impl/CouponServiceImpl.java

```diff
@@ -83,28 +84,31 @@ public CouponDTO findCouponById(UUID id) {
 
     @Transactional
     @Override
-    public CouponDTO assignCouponToUser(CreateCouponDTO createCouponDTO) {
-        log.debug("Finding of certain user by ID");
-
+    public CouponDTO assignCouponToUser(UUID discountId) {
+        log.debug(String.format("Check: if user already have Favorite of Discount with ID %s ", discountId));
+        String userEmail = SecurityContextHolder.getContext().getAuthentication().getName();
+        if (couponRepository.findByDiscountIdAndAndUserEmail(discountId, userEmail).isPresent()) {
+            throw new CreationRestrictedException(String
+                    .format("Coupon for Discount with id %s does already exist", discountId));
+        }
```

Comment:
You can use existsByDiscountIdAndUserEmail, it'll return boolean

Reply:
the same